### PR TITLE
Add a new extension: sphinx.ext.imgconverter

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Feature

### Purpose
This extension converts images in your document to appropriate format for builders.
For example, it allows you to use SVG images with LaTeX builder.
As a result, you don't mind what image format the builder supports.

Internally, this extension uses Imagemagick to convert images.

Imagemagick: https://www.imagemagick.org/script/index.php

### Relates
- #2859

### CODE REVIEW

File: doc/ext/imgconverter.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
+.. highlight:: rest
+
+:mod:`sphinx.ext.imgconverter` -- Convert images to appropriate format for builders
```
Comment (tk0miya) [inline doc/ext/imgconverter.rst:3]: I need to advice the module name. Is it better to use `imagemagick` for its name?

Comment (shimizukawa) [reply doc/ext/imgconverter.rst:3]: For now, this extension seems it depends to imagemagick command line interface, so this extension seems to be for imagemagick.

File: sphinx/ext/imgconverter.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            ret = subprocess.call([self.config.image_converter, '-version'],
+                                  stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+            if ret == 0:
+                return True
+            else:
+                return False
+        except (OSError, IOError):
+            logger.warning(_('convert command %r cannot be run.'
+                             'check the image_converter setting'),
+                           self.config.image_converter)
```
Comment (shimizukawa) [inline sphinx/ext/imgconverter.py:48]: Question is: how does user know what command (imagemagick) is required to use this extension?

File: sphinx/ext/imgconverter.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+class ImagemagickConverter(ImageConverter):
+    conversion_rules = [
+        ('image/svg+xml', 'image/png'),
+        ('application/pdf', 'image/png'),
+    ]
+
+    def is_available(self):
+        # type: () -> bool
+        """Confirms the converter is available or not."""
+        try:
```
Comment (shimizukawa) [inline sphinx/ext/imgconverter.py:37]: I think debug log is useful to know what command is executed here.

File: sphinx/ext/imgconverter.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        except (OSError, IOError):
+            logger.warning(_('convert command %r cannot be run.'
+                             'check the image_converter setting'),
+                           self.config.image_converter)
+            return False
+
+    def convert(self, _from, _to):
+        # type: (unicode, unicode) -> bool
+        """Converts the image to expected one."""
+        try:
```
Comment (shimizukawa) [inline sphinx/ext/imgconverter.py:54]: I think debug log is useful to know what command is executed here.

**Review by shimizukawa — COMMENTED** [review]
Thanks for the converting implementation! It looks great feature!

Implementation looks good to me.
I commented about naming and logging.

File: doc/ext/imgconverter.rst (hunk 1 above)
Comment (tk0miya) [reply doc/ext/imgconverter.rst:3]: Right. So I need better naming for this. What about `sphinx.ext.imgconverter.imagemagick`?

File: sphinx/ext/imgconverter.py (hunk 3 above)
Comment (tk0miya) [reply sphinx/ext/imgconverter.py:37]: done

File: sphinx/ext/imgconverter.py (hunk 4 above)
Comment (tk0miya) [reply sphinx/ext/imgconverter.py:54]: done

File: sphinx/ext/imgconverter.py (hunk 2 above)
Comment (tk0miya) [reply sphinx/ext/imgconverter.py:48]: only from docs. any idea?

File: doc/ext/imgconverter.rst (hunk 1 above)
Comment (tk0miya) [reply doc/ext/imgconverter.rst:3]: As a result of discussion in hack-day, we agreed with current name: `sphinx.ext.imgconverter`.

**Pull request merged**